File: frontend/src/features/achievements/types/achievement.ts
export interface Achievement {
  id: string;
  name: string;
  description?: string | null;
  trigger: "metric" | "api" | "streak";
  badgeUrl?: string | null;
  progress?: number;
  rarity?: number;
}

export interface UserAchievement extends Achievement {
  achievedAt: string | null;
}

File: backend/Data/Models/Achievement.cs
namespace Backend.Data.Models;

public class Achievement
{
    public int Id { get; set; }
    public string Key { get; set; } = null!; 
    public string Name { get; set; } = null!;
    public string Description { get; set; } = null!;
    public string Trigger { get; set; } = null!;
    public int Threshold { get; set; }
}